classes/OpeningHours/Module/Shortcode/IsOpen.php

```diff
@@ -81,12 +83,40 @@ public function shortcode ( array $attributes ) {
 
     $attributes['is_open'] = $isOpen;
     $attributes['classes'] .= ($isOpen) ? $attributes['open_class'] : $attributes['closed_class'];
-    $attributes['text'] = ($isOpen) ? $attributes['open_text'] : $attributes['closed_text'];
+
+    // If the attribute show_closed_holidays is enabled
+    if ($attributes['show_closed_holidays']) {
+        $holidaysList = $this->getTodaysHolidaysCommaSeperated($todayData);
+        $closedText = $holidaysList ? sprintf($attributes['closed_holiday_text'], $holidaysList) : $attributes['closed_text'];
+    } else {
+        $closedText = $attributes['closed_text'];
+    }
+
+    $attributes['text'] = ($isOpen) ? $attributes['open_text'] : $closedText;
 
     echo $this->renderShortcodeTemplate($attributes, 'shortcode/is-open.php');
   }
 
   /**
+   * Retrieves holiday names for today
+   * @param  array $todayData   Data for today
+   * @return string            Extracted holiday name(s)
+   */
+  public function getTodaysHolidaysCommaSeperated($todayData) {
+      if (count($todayData['holidays']) > 0) {
+        $holidayNames = '';
+
+        foreach ($todayData['holidays'] as $holiday) {
+            $holidayNames .= $holiday->getName() . ', ';
```

Comment:
`implode` was fine. I was just wondering why we need the `trim(rtrim(...))`. `implode` inserts the glue only between the elements and not around. So the only case when `trim` actually has an effect would be when a holiday name starts or ends with a comma, but then the "least surprise" for the user would be to have the comma in the message

Reply:
Gah yes your right.